python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -1223,8 +1245,451 @@ def asset_check_spec(self) -> AssetCheckSpec:
+
+    # IContext interface methods
+
+    @property
+    def is_asset_step(self) -> bool:
+        return self.op_execution_context.has_assets_def
+
+    @public
+    @property
+    def asset_key(self) -> AssetKey:
+        return self._op_execution_context.asset_key
+
+    @property
+    def asset_keys(self) -> Sequence[AssetKey]:
+        return self._asset_keys
+
+    @property
+    def provenance(self) -> Optional[DataProvenance]:
+        return self._provenance_by_asset_key[self.asset_key]
+
+    @property
+    def provenance_by_asset_key(self) -> Mapping[AssetKey, Optional[DataProvenance]]:
+        return self._provenance_by_asset_key
+
+    @property
+    def code_version(self) -> Optional[str]:
+        return self.code_version_by_asset_key[self.asset_key]
+
+    @property
+    def code_version_by_asset_key(self) -> Mapping[AssetKey, Optional[str]]:
+        return self._code_version_by_asset_key
+
+    @public
+    @property
+    def is_partition_step(self) -> bool:
+        return self._op_execution_context.has_partition_key
+
+    @property
+    def partition_key(self) -> str:
+        return self.op_execution_context.partition_key
+
+    @public
+    @property
+    def partition_key_range(self) -> PartitionKeyRange:
+        return self._op_execution_context.partition_key_range
+
+    @property
+    def partition_time_window(self) -> TimeWindow:
+        return self.op_execution_context.partition_time_window
+
+    @public
+    @property
+    def run_id(self) -> str:
+        return self._op_execution_context.run_id
+
+    @property
+    def job_name(self) -> Optional[str]:
+        return self.op_execution_context.job_name
+
+    @property
+    def retry_number(self) -> int:
+        return self.op_execution_context.retry_number
+
+    # Additional methods
+
+    @public
+    @property
+    def instance(self) -> DagsterInstance:
+        """DagsterInstance: The current Dagster instance."""
+        return self._op_execution_context.instance
+
+    @public
+    @property
+    def dagster_run(self) -> DagsterRun:
+        """PipelineRun: The current pipeline run."""
+        return self._op_execution_context.dagster_run
+
+    @public
+    @property
+    def pdb(self) -> ForkedPdb:
+        return self._op_execution_context.pdb
+
+    @public
+    @property
+    def log(self) -> DagsterLogManager:
+        """DagsterLogManager: The log manager available in the execution context."""
+        return self._op_execution_context.log
+
+    @public
+    def log_event(self, event: UserEvent) -> None:
+        return self._op_execution_context.log_event(event)
+
+    @public
+    @property
+    def assets_def(self) -> AssetsDefinition:
+        return self._assets_def
+
+    @public
+    @property
+    def selected_asset_keys(self) -> AbstractSet[AssetKey]:
+        return self._selected_asset_keys
+
+    # TODO - both get_asset_provenance and get_assets_code_version query the instance - do we want to
```

Comment:
Do we want to keep `get_asset_code_version` and `get_asset_provenance`? they would allow users to get the code version/provenance for ay asset (not just those in the assets definition), but require instance queries. 

I think i'm inclined to remove them and add back if we get requests, but not strongly tied to that

Reply:
sean suggested switching to a `fetch` prefix instead of `get` https://github.com/dagster-io/dagster/pull/16596#discussion_r1333131217 so that is another option

Reply:
`OpExecutionContext.get_asset_provenance` is used to create the ext context [here](https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_core/ext/context.py#L137), used in the init of the AssetExecutionContext in this PR, in an asset in the toys project, and in a test for data versioning.